storm-core/src/clj/backtype/storm/daemon/executor.clj

```diff
@@ -526,7 +526,8 @@
                                                                         out-tuple
                                                                         overflow-buffer)
                                                            ))
-                                         (if rooted?
+                                         (if (and rooted?
+                                                  (seq out-ids)) ;; not empty
```

Comment:
`(not (.isEmpty out-ids))` might be better. 
`seq` is going to create a new ISeq and do copying every execution of this loop. 
Probably not a big deal, but it's easy to make better.